DNN Platform/Tests/DotNetNuke.Tests.Web/DependencyInjection/BuildUpExtensionsTests.cs

```diff
@@ -0,0 +1,162 @@
+        }
+
+        [Test]
+        public void BuildUp_PrivateProperty_Test()
+        {
+            this.MockCBO<PrivateFilterAttribute>();
+            var filter = new PrivateFilterAttribute();
+
+            this.container.BuildUp(filter);
+            filter.OnActionExecuted(null);
+
+            this.VerifyEventLoggerInvoked();
+        }
+
+        [Test]
+        public void BuildUp_ProtectedProperty_Test()
+        {
+            this.MockCBO<ProtectedFilterAttribute>();
+            var filter = new ProtectedFilterAttribute();
+
+            this.container.BuildUp(filter);
+            filter.OnActionExecuted(null);
+
+            this.VerifyEventLoggerInvoked();
+
+        }
+
+        [Test]
+        public void BuildUp_ProtectedInternalProperty_Test()
+        {
+            this.MockCBO<ProtectedInternalFilterAttribute>();
+            var filter = new ProtectedInternalFilterAttribute();
+
+            this.container.BuildUp(filter);
+            filter.OnActionExecuted(null);
+
+            this.VerifyEventLoggerInvoked();
+        }
+
+        [Test]
+        public void BuildUp_InternalProperty_Test()
+        {
+            this.MockCBO<InternalFilterAttribute>();
+            var filter = new InternalFilterAttribute();
+
+            this.container.BuildUp(filter);
+            filter.OnActionExecuted(null);
+
+            this.VerifyEventLoggerInvoked();
+        }
+
+        [Test]
+        public void BuildUp_NoDependencyAttribute_Test()
+        {
+            this.MockCBO<PublicNoDependencyAttributeFilterAttribute>();
+            var filter = new PublicNoDependencyAttributeFilterAttribute();
+
+            this.container.BuildUp(filter);
+            Assert.Throws<NullReferenceException>(() => filter.OnActionExecuted(null));
+
+            this.VerifyEventLoggerInvoked(Times.Never());
+        }
+
+        [Test]
+        public void BuildUp_NoSetter_Test()
+        {
+            this.MockCBO<NoSetFilterAttribute>();
+            var filter = new NoSetFilterAttribute();
+
+            this.container.BuildUp(filter);
+            Assert.Throws<NullReferenceException>(() => filter.OnActionExecuted(null));
+
+            this.VerifyEventLoggerInvoked(Times.Never());
+        }
+
+        void MockCBO<T>()
+            where T : IFilter
+        {
+            var properties = typeof(T)
+                .GetProperties(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic)
+                .Where(propertyInfo => propertyInfo.GetSetMethod(true) != null && propertyInfo.GetCustomAttribute<DependencyAttribute>() != null)
+                .ToList();
+
+            var mockCbo = new Mock<ICBO>();
+            mockCbo
+                .Setup(x => x.GetCachedObject<IEnumerable<PropertyInfo>>(It.IsAny<CacheItemArgs>(), It.IsAny<CacheItemExpiredCallback>(), It.IsAny<bool>()))
+                .Returns(properties);
+
+            CBO.SetTestableInstance(mockCbo.Object);
+        }
+
+
```

Comment:
We don't need to duplicate the logic from the production code in order to test with the cache, we just need to make sure the callback gets used.  I guess this can be moved into the `SetUp`, since it no longer depends on the type being retrieved.
```suggestion
            var mockCbo = new Mock<ICBO>();
            mockCbo
                .Setup(x => x.GetCachedObject<IEnumerable<PropertyInfo>>(It.IsAny<CacheItemArgs>(), It.IsAny<CacheItemExpiredCallback>(), It.IsAny<bool>()))
                .Returns<CacheItemArgs, CacheItemExpiredCallback, bool>((args, callback, _) => (IEnumerable<PropertyInfo>)callback(args));

            CBO.SetTestableInstance(mockCbo.Object);
        }

```

Reply:
🤦‍♂️ I knew there was a way to do this but couldn't figure it out. Thanks for the tips, I'll update the unit tests